prow/apis/prowjobs/v1/types.go

```diff
@@ -156,7 +157,7 @@ type ProwJobSpec struct {
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
 
 	// RerunAuthConfig holds information about which users can rerun the job
-	RerunAuthConfig RerunAuthConfig `json:"rerun_auth_config,omitempty"`
+	RerunAuthConfig *RerunAuthConfig `json:"rerun_auth_config,omitempty"`
```

Comment:
It doesn't need to be. It seems more efficient and cleaner; also, with the added benefit of omittance from serialized PJ object.

Reply:
I do like avoiding the serialization for when I have to stare at ProwJob CRDs, but we probably don't need to optimize for that. I don't think that users spend enough time looking at PJ CRDs for one extra line of YAML to be much of a concern.
We traditionally avoid pointer types unless there is a reasonable improvement in performance. It is real hard to accidentally segfault if you don't have pointers 😎 

I think thats a lame stance though. If you get a segfault in prod there is some bug that wasn't caught by tests. Avoiding pointers to avoid segfaults just replaces nil with an empty struct, preventing the bug from crashing the program immediately, but making it more subtle. If we start seeing problems with segfaults in prod I think we should improve our testing patterns, not stop using pointers. If you want to keep the pointer type just make sure you have unit test coverage for the nil cases.

Reply:
I appreciate the detailed explanation and I agree with your stance. I will comb through the unit tests I wrote/changed that to ensure _all_ these cases are covered.

Reply:
Updated: using nil safe receivers backed by unit tests.